File: app/src/main/java/fr/nihilus/xenobladechronicles/datastore/AppDatabase.java
package fr.nihilus.xenobladechronicles.datastore;

import android.arch.persistence.room.Database;
import android.arch.persistence.room.RoomDatabase;
import android.arch.persistence.room.TypeConverters;

import fr.nihilus.xenobladechronicles.model.Monster;

@Database(entities = {Monster.class}, version = 1)
@TypeConverters({Converters.class})
public abstract class AppDatabase extends RoomDatabase {
    public static final String DATABASE_NAME = "xenoblade.db";

    public abstract MonsterDao monsterDao();
}

File: app/src/main/java/fr/nihilus/xenobladechronicles/datastore/AppDatabase.kt
package fr.nihilus.xenobladechronicles.datastore

import androidx.room.Database
import androidx.room.RoomDatabase
import androidx.room.TypeConverters
import fr.nihilus.xenobladechronicles.model.Monster

@Database(entities = [Monster::class], version = 1)
@TypeConverters(Converters::class)
abstract class AppDatabase : RoomDatabase() {
    abstract val monsterDao: MonsterDao

    companion object {
        const val DATABASE_NAME = "xenoblade.db"
    }
}
